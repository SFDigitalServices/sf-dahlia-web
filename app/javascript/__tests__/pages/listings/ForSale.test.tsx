import React from "react"
import { render, cleanup } from "@testing-library/react"
import ForSale from "../../../pages/listings/for-sale"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"

const axios = require("axios")

jest.mock("axios")

jest.mock("react-helmet-async", () => {
  return {
    HelmetProvider: ({ children }: { children: React.ReactNode }) => <div>{children}</div>, // Mock HelmetProvider
    Helmet: ({ children }: { children: React.ReactNode }) => <div>{children}</div>, // Mock Helmet component
  }
})

describe("For Sale", () => {
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
    jest.resetAllMocks()
  })

  it("renders ForSale component", async (done) => {
    axios.get.mockResolvedValue({ data: { listings: [] } })

    const { findByText, asFragment } = render(<ForSale assetPaths="/" />)

    expect(await findByText("Buy affordable housing")).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
    done()
  })

  it("listings with multiple listings render the first image in the array", async (done) => {
    axios.get.mockResolvedValue({ data: { listings: [openSaleListing] } })

    const { findByAltText } = render(<ForSale assetPaths="/" />)

    const image = await findByAltText(`${openSaleListing.Building_Name} Building`)
    expect(image.getAttribute("src")).toBe(openSaleListing.Listing_Images[0].Image_URL)
    done()
  })
})
