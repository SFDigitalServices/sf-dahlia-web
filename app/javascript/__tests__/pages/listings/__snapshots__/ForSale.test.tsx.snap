// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`For Sale renders ForSale component 1`] = `
<DocumentFragment>
  <div
    class="notranslate site-wrapper"
  >
    <div
      class="site-content"
    >
      <title
        class="notranslate"
      >
        Sale Listings
      </title>
      <meta
        charset="utf-8"
      />
      <meta
        content="IE=edge"
        http-equiv="X-UA-Compatible"
      />
      <meta
        content="width=device-width, initial-scale=1"
        name="viewport"
      />
      <meta
        content="M1zcm6GGM6sHSF_jvkq254DbYAj94JYbFC7ArZDAXlg"
        name="google-site-verification"
      />
      <meta
        content="http://localhost/"
        property="og:url"
      />
      <meta
        content="website"
        property="og:type"
      />
      <meta
        content="Sale Listings"
        property="og:title"
      />
      <meta
        content="DAHLIA San Francisco Housing Portal"
        property="og:description"
      />
      <meta
        content="DAHLIA San Francisco Housing Portal"
        name="description"
      />
      <meta
        property="og:image"
      />
      <meta
        content="DAHLIA San Francisco Housing Portal"
        property="og:site_name"
      />
      <meta
        content="summary_large_image"
        name="twitter:card"
      />
      <header
        class="site-header"
      >
        <a
          class="site-header__skip-link"
          href="#main-content"
        >
          Skip to main content
        </a>
        <div
          class="language-bar"
        >
          <div
            class="language-bar__inner"
          >
            <nav
              aria-label="Choose your language"
              class="language-nav"
            >
              <ul
                class="language-nav__list"
              >
                <li>
                  <button
                    class="language-nav__list-button is-active"
                    type="button"
                  >
                    English
                  </button>
                </li>
                <li>
                  <button
                    class="language-nav__list-button"
                    type="button"
                  >
                    Español
                  </button>
                </li>
                <li>
                  <button
                    class="language-nav__list-button"
                    type="button"
                  >
                    中文
                  </button>
                </li>
                <li>
                  <button
                    class="language-nav__list-button"
                    type="button"
                  >
                    Filipino
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
        <div
          class="site-header__notice "
        >
          <div
            class="site-header__notice-text"
          >
            <span
              class="feedback-link"
            >
              We'd love to get 
              <a
                href="https://airtable.com/shrw64DubWTQfRkdo"
                target="_blank"
              >
                your feedback.
              </a>
            </span>
          </div>
        </div>
        <nav
          aria-label="main navigation"
          class="site-header__container"
          role="navigation"
        >
          <div
            class="site-header__base  site-header__base-wrapped"
          >
            <div
              class="site-header__logo-container site-header__logo-width-med"
              id="site-header-logo"
            >
              <a
                aria-label="DAHLIA San Francisco Housing Portal"
                class="site-header__logo translate site-header__custom-width"
                href="/"
                target="_self"
              >
                <div
                  class="site-header__logo-content site-header__image-only-container"
                >
                  <img
                    alt="Site logo"
                    class="site-header__logo-image site-header__image-only"
                  />
                </div>
              </a>
            </div>
            <div
              class="site-header__navbar-menu site-header__navbar-wrapped"
              id="site-header-links"
            >
              <a
                class="site-header__link  "
                data-testid="Rent-0"
                href="/listings/for-rent"
                target="_self"
              >
                <div>
                  Rent
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="Buy-1"
                href="/listings/for-sale"
                target="_self"
              >
                <div>
                  Buy
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="My Favorites-2"
                href="/favorites"
                target="_self"
              >
                <div>
                  My Favorites
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="Get Assistance-3"
                href="/get-assistance"
                target="_self"
              >
                <div>
                  Get Assistance
                </div>
              </a>
              <a
                class="site-header__link  "
                data-testid="Sign In-4"
                href="/sign-in"
                target="_self"
              >
                <div>
                  Sign In
                </div>
              </a>
            </div>
          </div>
        </nav>
      </header>
      <main
        data-testid="main-content-test-id"
        id="main-content"
      >
        <div>
          <div
            id="page-header"
          >
            <header
              class="page-header buy-header"
            >
              <hgroup
                class="page-header__group"
              >
                <div
                  class="buy-header_columns"
                >
                  <h1
                    class="buy-header_title buy-header_left_col"
                  >
                    Buy affordable housing
                  </h1>
                  <div
                    class="mb-8 buy-header_right_col"
                  >
                    <a
                      class="button is-primary is-fullwidth"
                      href="#nav-bar-container"
                    >
                      See homes for sale
                    </a>
                  </div>
                  <div
                    class="buy-header_left_col"
                  >
                    <div
                      class="mb-8"
                    >
                      <h2
                        class="text__underline-weighted mb-5 text-gray-750"
                      >
                        Before applying
                      </h2>
                      <p
                        class="text-gray-750"
                      >
                        Make sure you:
                      </p>
                      <ol
                        class="numbered-list my-5"
                      >
                        <li
                          class="text-gray-750 primary-lighter-markup-link"
                        >
                          Haven't owned residential property in the past 3 years
                        </li>
                        <li
                          class="text-gray-750 primary-lighter-markup-link"
                        >
                          <span>
                            Meet our 
                            <a
                              href="https://sf.gov/determine-if-you-can-buy-affordable-housing-program"
                              target="_blank"
                            >
                              income requirements
                            </a>
                          </span>
                        </li>
                        <li
                          class="text-gray-750 primary-lighter-markup-link"
                        >
                          <span>
                            Complete 
                            <a
                              href="https://sf.gov/sign-complete-homebuyer-education"
                              target="_blank"
                            >
                              homebuyer education
                            </a>
                          </span>
                        </li>
                        <li
                          class="text-gray-750 primary-lighter-markup-link"
                        >
                          <span>
                            Get pre-approved for a mortgage loan by a 
                            <a
                              href="https://sf.gov/reports/october-2023/find-lender-below-market-rate-program"
                              target="_blank"
                            >
                              lender on our list
                            </a>
                          </span>
                        </li>
                        <li
                          class="text-gray-750 primary-lighter-markup-link"
                        >
                          Have enough in savings for closing costs and downpayment
                        </li>
                      </ol>
                      <div
                        class="text-gray-750 primary-lighter-markup-link-desktop"
                      >
                        <span>
                          Read 
                          <a
                            href="https://sf.gov/determine-if-you-can-buy-affordable-housing-program"
                            target="_blank"
                          >
                            our full list of requirements
                          </a>
                           for more details.
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="buy-header_right_col"
                  >
                    <div
                      class="md:bg-white md:p-4"
                    >
                      <h2
                        class="text__underline-weighted mb-5"
                      >
                        Get a down payment loan
                      </h2>
                      <p
                        class="mb-4"
                      >
                        Get a loan up to $500,000 to buy a home through the 2025 Downpayment Assistance Loan Program (DALP).
                      </p>
                      <p
                        class="font-bold"
                      >
                        Application dates
                      </p>
                      <p
                        class="mb-4"
                      >
                        Apply between March 4 and June 2, 2025
                      </p>
                      <p
                        class="mb-4"
                      >
                        <a
                          href="https://www.sf.gov/apply-downpayment-loan-buy-market-rate-home"
                          target="_blank"
                        >
                          Learn more about DALP and how to apply
                        </a>
                      </p>
                    </div>
                  </div>
                </div>
              </hgroup>
            </header>
          </div>
          <div
            class="directory-page-navigation-bar__container"
            id="nav-bar-container"
          >
            <div
              class="directory-page-navigation-bar"
            >
              <a
                class="seeds-button directory-page-navigation-bar__button"
                href="#enter-a-lottery"
                id="nav-bar-button-enter-a-lottery"
              >
                <span
                  class="ui-icon ui-medium"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 51 45"
                  >
                    <path
                      d="M49.7173 21.1504C50.2904 21.6532 50.3425 22.4593 49.7608 23.1094C49.3353 23.6815 48.4496 23.7335 47.8766 23.1528L44.6292 20.2923V37.4466C44.6292 41.278 41.5207 44.3812 37.6829 44.3812H12.6762C8.84011 44.3812 5.72991 41.278 5.72991 37.4466V20.2923L2.48165 23.1528C1.90597 23.7335 1.02805 23.6815 0.520446 23.1094C0.0127579 22.4593 0.0676337 21.6532 0.642961 21.1504L24.2592 0.346902C24.7888 -0.115634 25.5703 -0.115634 26.0999 0.346902L49.7173 21.1504ZM12.6762 41.6074H18.2332V27.7383C18.2332 26.204 19.4749 24.9644 21.0118 24.9644H29.3473C30.8842 24.9644 32.1258 26.204 32.1258 27.7383V41.6074H37.6829C39.9838 41.6074 41.8507 39.7437 41.8507 37.4466V17.9172L25.1795 3.23671L8.50843 17.9172V37.4466C8.50843 39.7437 10.3752 41.6074 12.6762 41.6074ZM21.0118 41.6074H29.3473V27.7383H21.0118V41.6074Z"
                    />
                  </svg>
                </span>
                Enter a lottery (0)
              </a>
              <a
                class="seeds-button directory-page-navigation-bar__button"
                href="#buy-now"
                id="nav-bar-button-buy-now"
              >
                <img
                  alt="House Circle Check"
                  height="16"
                  width="16"
                />
                Buy now (0)
              </a>
              <a
                class="seeds-button directory-page-navigation-bar__button"
                href="#upcoming-lotteries"
                id="nav-bar-button-upcoming-lotteries"
              >
                <span
                  class="ui-icon ui-medium"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 32 32"
                  >
                    <path
                      d="M23.258 15.994h-7.251v-11.266c0-0.368-0.299-0.667-0.667-0.667s-0.667 0.299-0.667 0.667v11.931c0 0.368 0.299 0.667 0.667 0.667h7.917c0.368 0 0.667-0.299 0.667-0.667s-0.298-0.667-0.667-0.667z"
                    />
                    <path
                      d="M16 0.333c-8.64 0-15.669 7.029-15.669 15.669s7.029 15.669 15.669 15.669c8.64 0 15.669-7.027 15.669-15.667s-7.029-15.67-15.669-15.67zM16 30.338c-7.906 0-14.336-6.43-14.336-14.334s6.43-14.338 14.336-14.338 14.336 6.43 14.336 14.336c0 7.904-6.43 14.336-14.336 14.336z"
                    />
                  </svg>
                </span>
                Upcoming lotteries (0)
              </a>
              <a
                class="seeds-button directory-page-navigation-bar__button"
                href="#lottery-results"
                id="nav-bar-button-lottery-results"
              >
                <span
                  class="ui-icon ui-medium"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 32 32"
                  >
                    <path
                      d="M22.754 22.68h-13.387c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667h13.389c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667z"
                    />
                    <path
                      d="M22.754 18.68h-13.387c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667h13.389c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667z"
                    />
                    <path
                      d="M22.754 14.68h-13.387c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667h13.389c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667z"
                    />
                    <path
                      d="M9.365 12.013h7.438c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667h-7.438c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667z"
                    />
                    <path
                      d="M27.526 4.877l-4.416-4.358c-0.216-0.213-0.514-0.506-1.072-0.506h-16.285c-0.947 0-1.717 0.781-1.717 1.741v26.014c0 0.205 0.098 0.387 0.246 0.51l3.142 3.36c0.126 0.134 0.302 0.211 0.486 0.211s0.362-0.077 0.486-0.211l3.506-3.747 3.629 3.875c0.251 0.269 0.722 0.269 0.973 0l3.629-3.875 3.629 3.875c0.126 0.134 0.301 0.21 0.483 0.211 0.002 0 0.002 0 0.003 0 0.182 0 0.357-0.075 0.483-0.206l3.117-3.27c0.117-0.123 0.184-0.29 0.184-0.459v-22.029c0.002-0.562-0.274-0.902-0.509-1.138zM26.102 5.347h-3.398v-3.355l3.398 3.355zM24.259 30.341l-3.634-3.88c-0.126-0.134-0.302-0.211-0.486-0.211s-0.362 0.077-0.486 0.211l-3.629 3.875-3.629-3.875c-0.251-0.269-0.722-0.269-0.973 0l-3.506 3.747-2.544-2.718v-25.736c0-0.221 0.174-0.408 0.382-0.408l15.616-0.010v4.677c0 0.368 0.299 0.667 0.667 0.667h4.667v21.094l-2.445 2.566z"
                    />
                  </svg>
                </span>
                Lottery results (0)
              </a>
            </div>
          </div>
          <div
            class="match-banner"
          />
          <div
            id="listing-results"
          >
            <div
              id="enter-a-lottery"
            >
              <div
                class="listings-group__header listings-group__customHeader"
              >
                <div
                  class="listings-group__content"
                >
                  <div
                    class="listings-group__icon"
                  >
                    <span
                      class="ui-icon ui-xlarge"
                    >
                      <svg
                        fill="currentColor"
                        viewBox="0 0 51 45"
                      >
                        <path
                          d="M49.7173 21.1504C50.2904 21.6532 50.3425 22.4593 49.7608 23.1094C49.3353 23.6815 48.4496 23.7335 47.8766 23.1528L44.6292 20.2923V37.4466C44.6292 41.278 41.5207 44.3812 37.6829 44.3812H12.6762C8.84011 44.3812 5.72991 41.278 5.72991 37.4466V20.2923L2.48165 23.1528C1.90597 23.7335 1.02805 23.6815 0.520446 23.1094C0.0127579 22.4593 0.0676337 21.6532 0.642961 21.1504L24.2592 0.346902C24.7888 -0.115634 25.5703 -0.115634 26.0999 0.346902L49.7173 21.1504ZM12.6762 41.6074H18.2332V27.7383C18.2332 26.204 19.4749 24.9644 21.0118 24.9644H29.3473C30.8842 24.9644 32.1258 26.204 32.1258 27.7383V41.6074H37.6829C39.9838 41.6074 41.8507 39.7437 41.8507 37.4466V17.9172L25.1795 3.23671L8.50843 17.9172V37.4466C8.50843 39.7437 10.3752 41.6074 12.6762 41.6074ZM21.0118 41.6074H29.3473V27.7383H21.0118V41.6074Z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div>
                    <h2
                      class="listings-group__title"
                    >
                      Enter a lottery
                    </h2>
                    <div
                      class="listings-group__info"
                    >
                      Enter a lottery to buy a home at a below-market-rate (BMR) price.
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="empty-listings-view"
              >
                <h1
                  class="text-heading-xl pb-3"
                >
                  0 open lotteries
                </h1>
                <div
                  class="empty-listings-view_content"
                >
                  <div>
                    <span
                      class="ui-icon ui-medium empty-state-icon"
                    >
                      <svg
                        fill="currentColor"
                        viewBox="0 0 51 39"
                      >
                        <path
                          d="M44.6431 0.564941H7.14307C3.69092 0.564941 0.893066 3.36279 0.893066 6.81494V31.8149C0.893066 35.2671 3.69092 38.0649 7.14307 38.0649H44.6431C48.0952 38.0649 50.8931 35.2671 50.8931 31.8149V6.81494C50.8931 3.36279 48.0903 0.564941 44.6431 0.564941ZM7.14307 3.68994H44.6431C46.3657 3.68994 47.7681 5.09229 47.7681 6.81494V10.3315L28.7056 24.6284C27.0454 25.8706 24.7378 25.8706 23.0776 24.6284L4.01807 10.3306V6.81494C4.01807 5.09619 5.42041 3.68994 7.14307 3.68994ZM47.7681 31.8149C47.7681 33.5376 46.3657 34.9399 44.6431 34.9399H7.14307C5.42041 34.9399 4.01807 33.5376 4.01807 31.8149V14.1489L21.2056 27.1274C22.5845 28.1636 24.2388 28.6919 25.8931 28.6919C27.5474 28.6919 29.2036 28.1626 30.5806 27.1274L47.7681 14.1489V31.8149Z"
                        />
                      </svg>
                    </span>
                    <span>
                      <a
                        href="https://confirmsubscription.com/h/y/C3BAFCD742D47910"
                        target="_blank"
                      >
                        Get an email
                      </a>
                       whenever a new listing is posted
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              id="buy-now"
            >
              <div
                class="listings-group__header listings-group__customHeader"
              >
                <div
                  class="listings-group__content"
                >
                  <div
                    class="listings-group__icon"
                  >
                    <img
                      alt="Icon Home Check"
                      height="40"
                      width="40"
                    />
                  </div>
                  <div>
                    <h2
                      class="listings-group__title"
                    >
                      Buy now
                    </h2>
                    <div
                      class="listings-group__info"
                    >
                      Buy a home at a below-market-rate (BMR) price without entering a lottery. First come, first served.
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="empty-listings-view"
              >
                <h1
                  class="text-heading-xl pb-3"
                >
                  0 open listings
                </h1>
                <div
                  class="empty-listings-view_content"
                >
                  <div>
                    <span
                      class="ui-icon ui-medium empty-state-icon"
                    >
                      <svg
                        fill="currentColor"
                        viewBox="0 0 51 39"
                      >
                        <path
                          d="M44.6431 0.564941H7.14307C3.69092 0.564941 0.893066 3.36279 0.893066 6.81494V31.8149C0.893066 35.2671 3.69092 38.0649 7.14307 38.0649H44.6431C48.0952 38.0649 50.8931 35.2671 50.8931 31.8149V6.81494C50.8931 3.36279 48.0903 0.564941 44.6431 0.564941ZM7.14307 3.68994H44.6431C46.3657 3.68994 47.7681 5.09229 47.7681 6.81494V10.3315L28.7056 24.6284C27.0454 25.8706 24.7378 25.8706 23.0776 24.6284L4.01807 10.3306V6.81494C4.01807 5.09619 5.42041 3.68994 7.14307 3.68994ZM47.7681 31.8149C47.7681 33.5376 46.3657 34.9399 44.6431 34.9399H7.14307C5.42041 34.9399 4.01807 33.5376 4.01807 31.8149V14.1489L21.2056 27.1274C22.5845 28.1636 24.2388 28.6919 25.8931 28.6919C27.5474 28.6919 29.2036 28.1626 30.5806 27.1274L47.7681 14.1489V31.8149Z"
                        />
                      </svg>
                    </span>
                    <span>
                      <a
                        href="https://confirmsubscription.com/h/y/C3BAFCD742D47910"
                        target="_blank"
                      >
                        Get an email
                      </a>
                       whenever a new listing is posted
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="bg-primary-darker sale-directory"
            >
              <div
                class="max-w-5xl mx-auto p-2 md:p-4"
              >
                <div
                  class="action-block  primary-darker action-block__inline"
                >
                  <div
                    class="action-block__head"
                  >
                    <div
                      class="action-block__header"
                    >
                      <h2
                        class=""
                      >
                        Find more listings on our website:
                      </h2>
                    </div>
                  </div>
                  <div
                    class="action-block__actions"
                  >
                    <a
                      class="button ml-8"
                      href="https://sfmohcd.org/current-listings-city-second-program"
                      target="_self"
                    >
                      City Second Loan Program
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="listings-group"
              id="upcoming-lotteries"
            >
              <div
                class="listings-group__header"
              >
                <div
                  class="listings-group__content"
                >
                  <div
                    class="listings-group__icon"
                  >
                    <span
                      class="ui-icon ui-xlarge"
                    >
                      <svg
                        fill="currentColor"
                        viewBox="0 0 32 32"
                      >
                        <path
                          d="M23.258 15.994h-7.251v-11.266c0-0.368-0.299-0.667-0.667-0.667s-0.667 0.299-0.667 0.667v11.931c0 0.368 0.299 0.667 0.667 0.667h7.917c0.368 0 0.667-0.299 0.667-0.667s-0.298-0.667-0.667-0.667z"
                        />
                        <path
                          d="M16 0.333c-8.64 0-15.669 7.029-15.669 15.669s7.029 15.669 15.669 15.669c8.64 0 15.669-7.027 15.669-15.667s-7.029-15.67-15.669-15.67zM16 30.338c-7.906 0-14.336-6.43-14.336-14.334s6.43-14.338 14.336-14.338 14.336 6.43 14.336 14.336c0 7.904-6.43 14.336-14.336 14.336z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="listings-group__header-group"
                  >
                    <h2
                      class="listings-group__title"
                    >
                      Upcoming Lotteries
                    </h2>
                    <div
                      class="listings-group__info"
                    >
                      The application deadline has passed on the following listings, and you can still get information on the lottery date and time.
                    </div>
                  </div>
                </div>
                <div
                  class="listings-group__button"
                >
                  <button
                    class="button w-full"
                    tabindex="0"
                  >
                    Show Upcoming Lotteries (0)
                  </button>
                </div>
              </div>
            </div>
            <div
              class="listings-group"
              id="lottery-results"
            >
              <div
                class="listings-group__header"
              >
                <div
                  class="listings-group__content"
                >
                  <div
                    class="listings-group__icon"
                  >
                    <span
                      class="ui-icon ui-xlarge"
                    >
                      <svg
                        fill="currentColor"
                        viewBox="0 0 32 32"
                      >
                        <path
                          d="M22.754 22.68h-13.387c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667h13.389c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667z"
                        />
                        <path
                          d="M22.754 18.68h-13.387c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667h13.389c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667z"
                        />
                        <path
                          d="M22.754 14.68h-13.387c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667h13.389c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667z"
                        />
                        <path
                          d="M9.365 12.013h7.438c0.368 0 0.667-0.299 0.667-0.667s-0.299-0.667-0.667-0.667h-7.438c-0.368 0-0.667 0.299-0.667 0.667s0.299 0.667 0.667 0.667z"
                        />
                        <path
                          d="M27.526 4.877l-4.416-4.358c-0.216-0.213-0.514-0.506-1.072-0.506h-16.285c-0.947 0-1.717 0.781-1.717 1.741v26.014c0 0.205 0.098 0.387 0.246 0.51l3.142 3.36c0.126 0.134 0.302 0.211 0.486 0.211s0.362-0.077 0.486-0.211l3.506-3.747 3.629 3.875c0.251 0.269 0.722 0.269 0.973 0l3.629-3.875 3.629 3.875c0.126 0.134 0.301 0.21 0.483 0.211 0.002 0 0.002 0 0.003 0 0.182 0 0.357-0.075 0.483-0.206l3.117-3.27c0.117-0.123 0.184-0.29 0.184-0.459v-22.029c0.002-0.562-0.274-0.902-0.509-1.138zM26.102 5.347h-3.398v-3.355l3.398 3.355zM24.259 30.341l-3.634-3.88c-0.126-0.134-0.302-0.211-0.486-0.211s-0.362 0.077-0.486 0.211l-3.629 3.875-3.629-3.875c-0.251-0.269-0.722-0.269-0.973 0l-3.506 3.747-2.544-2.718v-25.736c0-0.221 0.174-0.408 0.382-0.408l15.616-0.010v4.677c0 0.368 0.299 0.667 0.667 0.667h4.667v21.094l-2.445 2.566z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="listings-group__header-group"
                  >
                    <h2
                      class="listings-group__title"
                    >
                      Lottery Results
                    </h2>
                    <div
                      class="listings-group__info"
                    >
                      You can now check lottery results on the remaining listings.
                    </div>
                  </div>
                </div>
                <div
                  class="listings-group__button"
                >
                  <button
                    class="button w-full"
                    tabindex="0"
                  >
                    Show Lottery Results (0)
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="action-block mt-4 primary-lighter action-block__block"
          >
            <div
              class="action-block__head"
            >
              <div
                class="action-block__icon"
              >
                <span
                  class="ui-icon ui-3xl"
                >
                  <svg
                    fill="transparent"
                    height="35"
                    viewBox="0 0 80 35"
                    width="80"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="transparent"
                      height="75"
                      width="80"
                    />
                    <g
                      clip-path="url(#clip0)"
                    >
                      <rect
                        fill="transparent"
                        height="1390"
                        transform="translate(-284 -530)"
                        width="1280"
                      />
                      <path
                        d="M61.3749 24.2402L43.2061 39.7424C41.8262 40.8341 39.7563 40.8341 38.3764 39.7424L19.9775 24.2402"
                        stroke="black"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-miterlimit="10"
                        stroke-width="2"
                      />
                      <path
                        d="M49.1858 41.271L61.375 50.878"
                        stroke="black"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-miterlimit="10"
                        stroke-width="2"
                      />
                      <path
                        d="M32.1668 41.271L19.9775 50.878"
                        stroke="black"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-miterlimit="10"
                        stroke-width="2"
                      />
                      <path
                        d="M14.458 24.0218V51.0961C14.458 53.9345 16.7579 56.1179 19.7477 56.1179H61.835C64.8248 56.1179 67.1247 53.9345 67.1247 51.0961V24.0218C67.1247 21.1834 64.8248 19 61.835 19H19.7477C16.7579 19 14.458 21.1834 14.458 24.0218Z"
                        stroke="black"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-miterlimit="10"
                        stroke-width="2"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0"
                      >
                        <rect
                          fill="white"
                          height="1390"
                          transform="translate(-284 -530)"
                          width="1280"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </span>
              </div>
              <div
                class="action-block__header"
              >
                <h2
                  class=""
                >
                  Get emailed whenever a new listing is posted.
                </h2>
              </div>
            </div>
            <div
              class="action-block__actions"
            >
              <a
                class="button"
                href="https://confirmsubscription.com/h/y/C3BAFCD742D47910"
                target="_blank"
              >
                Sign Up today
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
    <footer
      class="site-footer"
    >
      <div
        class="footer-row "
      >
        <div
          class="footer-row__section pb-8"
        >
          <img
            alt=""
            data-testid="footer-logo-test-id"
          />
        </div>
      </div>
      <div
        class="footer-row "
      >
        <div
          class="footer-row__section py-2"
        >
          <p
            class="text-gray-500"
          >
            <span>
              DAHLIA: San Francisco Housing Portal is a project of the 
              <br />
              <a
                class="text-white"
                href="https://sf.gov/departments/mayors-office-housing-and-community-development"
                target="_blank"
              >
                Mayor's Office of Housing and Community Development
              </a>
            </span>
          </p>
          <p
            class="text-xs mt-4 text-gray-500"
          >
            <span>
              in partnership with
              <br />
              <a
                class="text-white"
                href="https://sf.gov/departments/city-administrator/digital-services"
                target="_blank"
              >
                San Francisco Digital Services
              </a>
              <br />
              <a
                class="text-white"
                href="https://sf.gov/departments/mayors-office-innovation"
                target="_blank"
              >
                Mayor's Office of Civic Innovation
              </a>
            </span>
          </p>
        </div>
      </div>
      <div
        class="footer-row "
      >
        <div
          class="footer-row__section pb-8"
        >
          <p
            class="text-sm"
          >
            For listing and application questions, please contact the agent displayed on the listing. 
            <br />
            For general program inquiries, you may call MOHCD at 415-701-5500.
          </p>
        </div>
      </div>
      <section
        class="footer-sock"
      >
        <div
          class="footer-sock__inner"
        >
          <p
            class="footer-copyright"
          >
            © City & County of San Francisco
          </p>
          <nav
            aria-label="Footer"
            class="footer-nav"
          >
            <a
              class="text-gray-500"
              href="https://airtable.com/shrw64DubWTQfRkdo"
              target="_blank"
            >
              Give Feedback
            </a>
            <a
              class="text-gray-500"
              href="mailto:sfhousinginfo@sfgov.org"
              target="_self"
            >
              Contact
            </a>
            <a
              class="text-gray-500"
              href="/disclaimer"
              target="_self"
            >
              Disclaimer
            </a>
            <a
              class="text-gray-500"
              href="/privacy"
              target="_self"
            >
              Privacy Policy
            </a>
          </nav>
        </div>
      </section>
    </footer>
  </div>
</DocumentFragment>
`;
